main.js

```diff
@@ -0,0 +1,84 @@
+var board = {
+  pegs: [[5, 4, 3, 2, 1],
+  [],
+  []],
+
+  moveDisc: function(startPeg, endPeg) {
+  board.pegs.map(peg => {
+    var pegLine = "---";
+    var newArr = [pegLine].concat(peg);
+    return newArr;
+  })
+  
+    var startPegArray = board.pegs[(startPeg-1)];
+    var endPegArray = board.pegs[(endPeg-1)];
+
+    var startPegDisc = startPegArray[startPegArray.length - 1];
+    var endPegDisc = endPegArray[endPegArray.length - 1];
+    
+    if (startPegDisc > endPegDisc) {
+      console.log('Sorry, this is an illegal move!');
+    } else if (endPegDisc === undefined || startPegDisc < endPegDisc) {
+      startPegArray.splice(startPegArray.length - 1, 1);
+      endPegArray.push(startPegDisc);
+      console.log(`The move is successful.`);
+    }
+    this.checkWinner();
+    this.printBoard();
+  },
+
+  checkWinner: function() {
+    var winner = [5, 4, 3, 2, 1];
+   
+    if (this.pegs[1].length === winner.length || this.pegs[2].length === winner.length) {
+      console.log(`You're a winner!`);
+      this.pegs[0].push(5, 4, 3, 2, 1)
+      this.pegs[1].splice(0, 5);
+      this.pegs[2].splice(0, 5);
+    };
+  },
+
+  printBoard: function() {
+    board.pegs.map(peg => {
+      var pegLine = "---";
+      var newArr = [pegLine].concat(peg);
+      console.log(newArr);
```

Comment:
You're still effectively treating this as a `forEach`. You'll want to store the new array in a variable and print that, not printing on each pass.